streams/src/main/java/org/apache/kafka/streams/state/internals/RecordConverter.java

```diff
@@ -23,9 +23,11 @@
 public interface RecordConverter {
     ConsumerRecord<byte[], byte[]> convert(final ConsumerRecord<byte[], byte[]> record);
 
-    @SuppressWarnings("deprecation")
-    static RecordConverter converter() {
-        return record -> {
+    final class RecordConverters {
+        private static final RecordConverter IDENTITY_INSTANCE = record -> record;
+
+        @SuppressWarnings("deprecation")
+        private static final RecordConverter RAW_TO_TIMESTAMED_INSTANCE = record -> {
```

Comment:
Cache the functions so we don't have to lean on the compiler's decision to cache it and avoid creating a new function every invocation.